ts/features/design-system/core/DSDynamicCardRotation.tsx

```diff
@@ -0,0 +1,459 @@
+/* eslint-disable functional/immutable-data */
+import { H6, IOColors, VSpacer, hexToRgba } from "@pagopa/io-app-design-system";
+import {
+  Canvas,
+  Color,
+  DiffRect,
+  Image,
+  LinearGradient,
+  Mask,
+  RoundedRect,
+  Circle as SkiaCircle,
+  Group as SkiaGroup,
+  RadialGradient as SkiaRadialGradient,
+  rect,
+  rrect,
+  useImage,
+  vec
+} from "@shopify/react-native-skia";
+import * as React from "react";
+import { useState } from "react";
+import {
+  ColorValue,
+  LayoutChangeEvent,
+  LayoutRectangle,
+  StyleSheet,
+  Text,
+  View,
+  ViewStyle
+} from "react-native";
+import Animated, {
+  Extrapolate,
+  SensorType,
+  interpolate,
+  useAnimatedReaction,
+  useAnimatedSensor,
+  useAnimatedStyle,
+  useDerivedValue,
+  useSharedValue,
+  withSpring
+} from "react-native-reanimated";
+import Svg, { Circle, Defs, RadialGradient, Stop } from "react-native-svg";
+
+type CardSize = {
+  width: LayoutRectangle["width"];
+  height: LayoutRectangle["height"];
+};
+
+type LightSize = {
+  value: LayoutRectangle["width"];
+};
+
+/* LIGHT
+   Visual parameters */
+const lightSizePercentage: ViewStyle["width"] = "90%";
+const lightScaleMultiplier: number = 1;
+const lightOpacity: ViewStyle["opacity"] = 0.9;
+const lightSkiaOpacity: number = 0.4;
+/* Percentage of visible light when it's near
+card boundaries */
+const visibleLightPercentage: number = 0.25;
+
+/* CARD
+   Visual parameters */
+const cardAspectRatio: ViewStyle["aspectRatio"] = 7 / 4;
+const cardBorderRadius: number = 24;
+const cardBorderWidth: number = 1;
+const cardBorderColor: ColorValue = IOColors["hanPurple-500"];
+const cardBorderHighlighted: ColorValue = IOColors.white;
+const cardBorderOpacity: number = 0.65;
+// Drivers' License
+const cardGradient: Array<Color> = ["#F4ACD5", "#FCE6F2"];
+// Flag
+// const flagDistanceFromEdge: number = 16;
+// const flagSize: number = 32;
+
+/* MOVEMENT
+   Spring config for the light movement */
+const springConfig = {
+  mass: 1,
+  damping: 50,
+  stiffness: 200,
+  overshootClamping: false
+};
+
+export const DSDynamicCardRotation = () => {
+  /* On first render, store the current device orientation
+  using quaternions */
+  const rotationSensor = useAnimatedSensor(SensorType.ROTATION);
+  const { qx, qy } = rotationSensor.sensor.value;
+  const initialQx = useSharedValue(0);
+  const initialQy = useSharedValue(0);
+
+  const skiaTranslateX = useSharedValue(0);
+  const skiaTranslateY = useSharedValue(0);
+
+  useAnimatedReaction(
+    () => rotationSensor.sensor.value,
+    s => {
+      initialQx.value = s.qx;
+      initialQy.value = s.qy;
+    },
+    []
+  );
+
+  /* Not all devices are in an initial flat position on a surface
+    (e.g. a table) then we use relative rotation values,
+    not absolute ones  */
+  const relativeQx = useDerivedValue(() => qx - initialQx.value);
+  const relativeQy = useDerivedValue(() => qy - initialQy.value);
```

Comment:
Roll and pitch are simpler to use than quaternions for this case. We do not need to do complex calculations with them.
Also, I think that the X axis is inverted

```suggestion
  const { roll: initialRoll, pitch: initialPitch } =
    rotationSensor.sensor.value;
  const roll = useSharedValue(0);
  const pitch = useSharedValue(0);

  const skiaTranslateX = useSharedValue(0);
  const skiaTranslateY = useSharedValue(0);

  useAnimatedReaction(
    () => rotationSensor.sensor.value,
    sensor => {
      roll.value = sensor.roll;
      pitch.value = sensor.pitch;
    },
    []
  );

  /* Not all devices are in an initial flat position on a surface
    (e.g. a table) then we use relative rotation values,
    not absolute ones  */
  const relativeRoll = useDerivedValue(() => -(initialRoll - roll.value));
  const relativePitch = useDerivedValue(() => initialPitch - pitch.value);
```

Reply:
@mastro993 Addressed in this commit. Life is a lot easier without quaternions 😄  → [`b856d28` (#5715)](https://github.com/pagopa/io-app/pull/5715/commits/b856d2816687ff6c2beb0be42a5f85ed48df7a93)